### PR TITLE
pin aws-sdk-go-v2 dependency versions for ClusterAuth presign breaking change

### LINKED ISSUE #1248: [Bug]: EKS ClusterAuth v1.3.0 kubeconfig results in Unauthorized
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Resource(s)

_No response_

### Resource MRs required to reproduce the bug

_No response_

### Steps to Reproduce

After creating EKS `Cluster`, create `ClusterAuth` with `writeConnectionSecretToRef` configured. Create provider-kubernetes ProviderConfig to reference `ClusterAuth` Secret, or retrieve `kubeconfig` from Secret to use manually. `ProviderConfig` configured with IRSA `credentials.source` and `assumeRoleChain`.

### What happened?

When utilizing provider-aws-eks `v1.3.0`, usage of the `kubeconfig` results in `cannot get object: failed to get API group resources: unable to retrieve the complete list of server APIs: apps/v1: Unauthorized`. 

Downgrade to provider-aws-eks `v1.2.0`, and `kubeconfig` is updated and access to the EKS cluster is available.

The same results are exhibited when accessing an EKS cluster provisioned with `v1.2.0`, which worked as expected, but after upgrading to `v1.3.0`, the `Unauthorized` error began.

### Relevant Error Output Snippet

_No response_

### Crossplane Version

1.15.1

### Provider Version

1.3.0

### Kubernetes Version

v1.29.2

### Kubernetes Distribution

EKS

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by erhancagirici**
### Description of your changes

Fixes #1248 
Recent patch version bumps at the following `aws-sdk-go-v2` modules seems to break the request signature of the token for the generated kubeconfig for `ClusterAuth` resource. 
```
github.com/aws/aws-sdk-go-v2
github.com/aws/aws-sdk-go-v2/internal/configsources
github.com/aws/aws-sdk-go-v2/internal/endpoints/v2
```

This change reverts the version bumps and pins the versions.
Adds a separate standalone example manifest for `ClusterAuth` resource and added an Uptest post-assert hook for testing the resulting kubeconfig, by executing simple kubectl commands to ensure the cluster authentication.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

- Tested manually with `ClusterAuth` MR example, that fails on 1.3.0
- With Uptest using the new post-assert check for the resulting kubeconfig

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by jeanduplessis** [comment]
Is it possible to add a test to catch a regression here in the future? 
Is ClusterAuth already uptestable and we should consider it a critical component that should always be tested before a release?

**Comment by erhancagirici** [comment]
> Is it possible to add a test to catch a regression here in the future? Is ClusterAuth already uptestable and we should consider it a critical component that should always be tested before a release?

ClusterAuth is already uptestable, and included in the `examples/v1beta1/cluster.yaml`. The MR actually gets created, switches to ready & synced state,  and the connection details are published. In order to catch the regression here, the resulting kubeconfig in the connection details should be consumed.

We might consider utilizing https://github.com/upbound/configuration-aws-eks, some test MRs utilizing provider-kubernetes or some post-test script utilizing `kubectl`. cc @ulucinar @sergenyalcin

**Comment by ulucinar** [comment]
Hi @erhancagirici,
We may consider utilizing an [uptest `post-assert-hook`](https://github.com/crossplane/uptest/?tab=readme-ov-file#hooks) for `ClusterAuth.eks` in which we run a script that:
- Dumps the kubeconfig from the K8s connection secret onto the file system
- Runs a `kubectl` command against that kubeconfig

Could you please also trigger an uptest on `ClusterAuth.eks` with these modifications? We may also consider adding the example manifest `examples/eks/v1beta1/clusterauth.yaml`.

**Comment by erhancagirici** [comment]
/test-examples="examples/eks/v1beta1/clusterauth.yaml"

**Comment by erhancagirici** [comment]
/test-examples="examples/eks/v1beta1/clusterauth.yaml"

File: examples/eks/v1beta1/testhooks/check-clusterauth.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+set -aeuo pipefail
+
+# SPDX-FileCopyrightText: 2024 The Crossplane Authors <https://crossplane.io>
+#
+# SPDX-License-Identifier: CC0-1.0
+
+echo "obtain kubeconfig from ClusterAuth connection secret"
+${KUBECTL} -n upbound-system get secret sample-eks-cluster-conn -o go-template='{{ .data.kubeconfig | base64decode }}' > sampleclusterkube
```
Comment (ulucinar) [inline examples/eks/v1beta1/testhooks/check-clusterauth.sh:9]: nit: We may consider [reading the namespace from the environment](https://github.com/crossplane-contrib/provider-upjet-aws/blob/4a2b3710bd93ad7ebfd2ee9375fc65d4551bccc7/Makefile#L191) but let's do this in a separate PR in which we implement this for other scripts as well (if there are any).

**Review by ulucinar — APPROVED** [review]
Thank you @erhancagirici, lgtm.

File: examples/eks/v1beta1/testhooks/check-clusterauth.sh (hunk 1 above)
Comment (erhancagirici) [reply examples/eks/v1beta1/testhooks/check-clusterauth.sh:9]: the namespace is actually coming from the `writeConnectionSecretToRef` section of the particular example we are using, it is not necessarily the `CROSSPLANE_NAMESPACE`. That is why I choose to hardcode it. Though, we can consider parametrizing the namespace name at the example's `writeConnectionSecretToRef` if you prefer.

**Comment by erhancagirici** [comment]
/test-examples="examples/eks/v1beta1/clusterauth.yaml"

**Comment by mbbush** [comment]
It looks like this was probably caused by https://github.com/aws/aws-sdk-go-v2/pull/2438. I see a bunch of other projects citing that PR as an issue too. I don't understand why yet.

**Pull request merged**